src/NServiceBus.AcceptanceTests/Routing/SubscriptionBehaviorExtensions.cs

```diff
@@ -2,25 +2,30 @@ namespace NServiceBus.AcceptanceTests.Routing
 {
     using System;
     using AcceptanceTesting;
+    using CriticalError = NServiceBus.CriticalError;
 
     static class SubscriptionBehaviorExtensions
     {
-        public static void OnEndpointSubscribed<TContext>(this EndpointConfiguration configuration, Action<SubscriptionEventArgs, TContext> action) where TContext : ScenarioContext
+        public static void OnEndpointSubscribed<TContext>(this EndpointConfiguration b, Action<SubscriptionEventArgs, TContext> action) where TContext : ScenarioContext
         {
-            configuration.Pipeline.Register(new SubscriptionBehavior<TContext>.Registration("NotifySubscriptionBehavior", builder =>
+            b.Pipeline.Register<SubscriptionBehavior<TContext>.Registration>();
+
+            b.RegisterComponents(c => c.ConfigureComponent(builder =>
             {
                 var context = builder.Build<TContext>();
-                return new SubscriptionBehavior<TContext>(action, context, MessageIntentEnum.Subscribe);
-            }));
+                return new SubscriptionBehavior<TContext>(action, context, builder.Build<CriticalError>(), MessageIntentEnum.Subscribe);
+            }, DependencyLifecycle.InstancePerCall));
```

Comment:
why did you add this? Afaik behaviors are cached anyway, so instance per call will not result in what you would expect when defining it?